### PR TITLE
control/controlclient: fix sending peer capmap changes

### PR DESCRIPTION
**Pull request by clairew**
Instead of just checking if a peer capmap is nil, also check if the previous state's capmap was nil.
Updates tailscale/corp#17516

### CODE REVIEW

**Review by sailorfrag — APPROVED** [review]
I believe this is safe because it only applies to patchifying `PeerChanged` and `MapCap` is unused in peers for all previous versions. It fixes the integration test that was failing. Please wait for @maisem to also review before merge.

File: control/controlclient/map.go
Comment (bradfitz) [inline control/controlclient/map.go:668]: But now that I see this code, this is wrong. If these are equal, it shouldn't be a diff.

File: control/controlclient/map_test.go
Comment (bradfitz) [inline control/controlclient/map_test.go:869]: Also add a test of CapMap not changing and thus no diff.

**Comment by bradfitz** [comment]
Let's try to find a better word than "improve" ... all changes are improvements or we wouldn't be making them :)

Maybe "fix CapMap handling in peerChangeDiff"

**Review by sailorfrag — APPROVED** [review]
The change is doing more than checking for nil now, let's update the commit message but otherwise I think this should be good

File: control/controlclient/map.go
Comment (bradfitz) [inline control/controlclient/map.go:671]: looks like all the other return paths returning a diff don't alias memory provided by the caller

(line 678 too)

Comment (sailorfrag) [reply control/controlclient/map.go:671]: Can I make that a follow on PR to unblock Claire? The existing code was already aliasing memory.

Comment (bradfitz) [reply control/controlclient/map.go:671]: k

**Review by bradfitz — APPROVED** [review]

File: control/controlclient/map.go
Comment (clairew) [reply control/controlclient/map.go:671]: @sailorfrag pointed out that since the value type for NodeCapMap is of type tailcfg.RawMessage, which is a string thus immutable. So - aliasing memory is ok?

Comment (sailorfrag) [reply control/controlclient/map.go:671]: The RawMessage yes, the map itself, no

**Pull request merged**
